.teamcity/_self/projects/WebApp.kt

```diff
@@ -571,12 +571,44 @@ object Translate : BuildType({
 		cleanCheckout = true
 	}
 
+	artifactRules = """
+		translate => translate
+	""".trimIndent()
+
 	steps {
 		bashNodeScript {
-			name = "Translate Build Steps"
+			name = "Prepare environment"
+			scriptContent = """
+				# Install modules
+				${_self.yarn_install_cmd}
+			"""
+			dockerImage = "%docker_image_e2e%"
+		}
+		bashNodeScript {
+			name = "Extract strings"
+			scriptContent = """
+				# Run script to extract strings from source code
+				yarn run translate
+
+				# Move `calypso-strings.pot` to artifacts directory
+				mkdir -p ./translate
+				mv public/calypso-strings.pot ./translate/
+			"""
+			dockerImage = "%docker_image_e2e%"
+		}
+		bashNodeScript {
+			name = "Build New Strings .pot"
 			scriptContent = """
-				echo "@todo: Translate Build Script"
+				# Clone GP LocalCI Client
+				git clone --single-branch --depth=1 https://github.com/Automattic/gp-localci-client.git
+
+				# Build `localci-new-strings.pot`
+				DEFAULT_BRANCH=trunk bash gp-localci-client/generate-new-strings-pot.sh "%teamcity.build.branch%" "${Settings.WpCalypso.paramRefs.buildVcsNumber}" "./translate"
```

Comment:
Do we need to define the GitHub auth tokens somewhere?

Reply:
`gp-localci-client` is a public repo, so I don't think we need to. Or do you mean something else?

Reply:
The repo mentions GH rate-limiting -- wasn't sure if that was a concern here: https://github.com/Automattic/gp-localci-client#authentication

Reply:
I think we can export them as env variables in the script, but we can also define them as configuration parameters in TeamCity and then utilize them in bash without revealing their values